### PR TITLE
Add TOC and proper noun section to challenge guide

### PR DESCRIPTION
**Pull request by erictleung**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [ ] Closes currently open issue (replace XXXX with an issue no): Closes #XXXX

#### Description

- Add table of contents summarizing challenge style guide
- Start proper noun section for how certain words should appear
- Add constraint on using double quotes for seed and description

### CODE REVIEW

**Comment by QuincyLarson** [comment]
@erictleung Nice addition. 

Our style guide says the following:

> "Also, front-end development (adjective form with a dash) is when you working on the front end (noun form with no dash). The same goes with the back end, full stack, and many other compound terms."

I did some research and determined that was the most grammatically correct way to worth things, and I recommend we use this here in this style guide, too.

**Comment by teeleek** [comment]
Okay

Puttasak insee

เมื่อ 13 ก.พ. 2017 03:32 "Quincy Larson" <notifications@github.com> เขียนว่า

**Comment by erictleung** [comment]
@QuincyLarson whoops. I must have missed that 😅 I'll update this later today. I got the idea to add it because I noticed it being used as "frontend" and "backend" in one of the newer challenges. I'll try to find the challenge I found this in and make those changes as well.

**Comment by erictleung** [comment]
@QuincyLarson I've made some additional updates. I've added your quote about how to use "front-end" correctly. Additionally, I know there's a convention (or at least I've felt there has been) on using double quotes in the challenge descriptions and seed code. If I've missed something, let me know.

File: seed/challenge-style-guide.md [hunk 1]
```diff
@@ -74,13 +95,15 @@ Here are specific formatting guidelines for challenge text and examples:
 - Multi-line code examples go in `<blockquote>` tags, and use the `<br>` tag to separate lines. For HTML examples, remember to use escape characters to represent the angle brackets
 - A single horizontal rules (`<hr>` tag) should separate the text discussing the challenge concept and the challenge instructions
 - Additional information in the form of a note should be formatted `<strong>Note</strong><br>Rest of note text...`
+- Use double quotes when possible
```
Comment (raisedadead) [inline seed/challenge-style-guide.md]:
```diff
- ...when possible
+ ...where applicable
```

File: seed/challenge-style-guide.md [hunk 2]
```diff
@@ -74,13 +95,15 @@ Here are specific formatting guidelines for challenge text and examples:
...
 
 ## Formatting seed code
 
 Here are specific formatting guidelines for the challenge seed code:
 
 - Use two spaces to indent
 - JavaScript statements end with a semicolon
+- Use double quotes when possible
```
Comment (raisedadead) [inline seed/challenge-style-guide.md]:
```diff
- ...when possible
+ ...where applicable
```

**Review by raisedadead — COMMENTED** [review]
@erictleung a minor change to make it more obligatory?

**Comment by erictleung** [comment]
@raisedadead thanks for the review! Yeah, that wording works better 👍

**Comment by raisedadead** [comment]
LGTM, thanks @erictleung!

**Pull request merged**